content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md

```diff
@@ -40,6 +40,7 @@ For both methods you need this infrastructure:
   the masters
 - Three machines that meet [kubeadm's minimum
   requirements](/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#before-you-begin) for the workers
+- A highly available load balancer that operates at Layer 4 (such as one provided by a cloud provider, or [HAProxy](http://www.haproxy.org/)) with it's own name in DNS. The load balancer should be configured to forward the apiserver port (default: 6443) on the load balancer to the apiserver port on all control plane nodes. Before using a cloud provider provided load balancer, double check the documentation for [NAT hairpinning](https://en.wikipedia.org/wiki/Hairpinning) support.
```

Comment:
we should probably mention HTTPS and health checks here.
@ereslibre @fabriziopandini how would you word that?

Reply:
I would say that even if it's a layer 4 LB it should have the ability to perform HTTPS health checks (even if the certificate chain is ignored). That way it will remove from the active pool unhealthy apiservers.